### ISSUE TITLE
Add support for .do (Stata)

### ISSUE THREAD

**Issue by norrissam** [sbarex/SourceCodeSyntaxHighlight#223]
Hi and thanks for this great plugin. I was wondering if it would be possible to support .do files (Stata) even just as plaintext. Thanks!

**Comment by haoranliu666** [comment]
Yeah, .do files and .ado files. Thanks!

**Comment by haoranliu666** [comment]
No worry, StataCorp will solve this issue 

https://www.statalist.org/forums/forum/general-stata-discussion/general/1694451-macos-ventura-quicklook-for-stata-do-files-and-ado-files

**Comment by sbarex** [comment]
Ok, so I suspend the implementation of this language.

**Comment by haoranliu666** [comment]
> Ok, so I suspend the implementation of this language.

But the StataCorp version doesn't have syntax, just plain text. If you can make a preview with syntax, that would be even better:) Thank you!

**Comment by sbarex** [comment]
No, `Highlight` the rendering engine do not handle the Stata syntax, so I can only show the code as plain text (with line numbers if you want).

**Comment by norrissam** [comment]
If it's easy to do I'd still appreciate adding support for .do and .ado files - there are lots of old versions of Stata out there that this update won't fix.

**Comment by StevenLi-DS** [comment]
@sbarex I'm having the same issue. I wonder if you could consider letting the plugin render unsupported file format just as a plain text. I read the doc, and it seems like this is only doable when a file does not an extension now. Or maybe link a file format to another and allow it to render an unknown format as a known one? For example, Stata also has a format .stjson, and it is really just a json file, and they like to make it unique somehow.

**Comment by StevenLi-DS** [comment]
@sbarex thank you for the fix. Am I missing something here? I saw there is already 2.1.18 on brew, but not when you click check for updates

<img width="344" alt="CS-8d8nzM8v@2x" src="https://user-images.githubusercontent.com/33796896/231274810-b05aa78d-7235-4b60-870c-1402c3184398.png">

**Comment by sbarex** [comment]
now the automatic update should work

**Comment by dreistein543** [comment]
I have upgraded to version 2.1.18 :
<img width="409" alt="image" src="https://user-images.githubusercontent.com/30745407/231981274-21138dcb-50e5-4101-922b-70c719960545.png">

however, preview function still not work for Stata do file :
<img width="575" alt="image" src="https://user-images.githubusercontent.com/30745407/231981632-7c136583-bd51-46b3-8440-61751e67e1a6.png">

have I missed something important?

**Comment by sbarex** [comment]
> I have upgraded to version 2.1.18 : <img alt="image" width="409" src="https://user-images.githubusercontent.com/30745407/231981274-21138dcb-50e5-4101-922b-70c719960545.png">
> 
> however, preview function still not work for Stata do file : <img alt="image" width="575" src="https://user-images.githubusercontent.com/30745407/231981632-7c136583-bd51-46b3-8440-61751e67e1a6.png">
> 
> have I missed something important?

It is possible that the format has been registered by Stata with a particular UTI. From the inquiry window it is possible to see which UTI is associated with the extension.

**Comment by haoranliu666** [comment]
<img width="783" alt="Screenshot 2023-04-14 at 13 31 32" src="https://user-images.githubusercontent.com/25248404/232116477-07b01320-8c4f-4629-ada6-1ed8e384f209.png">
Yes, seems Stata registered the UTI and it's not working.

**Comment by sbarex** [comment]
I do not see the full uti in the screenshot

**Comment by sbarex** [comment]
Implemented on release 2.1.19 (68).
